src/compiler/llvm-to-backend/ptx/LLVMToPtx.cpp

```diff
@@ -70,15 +70,16 @@ class LibdevicePath {
     std::vector<std::string> SubDir {"nvvm", "libdevice"};
     std::string BitcodeDir = common::filesystem::join_path(CUDAPath, SubDir);
 
-    try {
-      auto Files = common::filesystem::list_regular_files(BitcodeDir);
-      for(const auto& F : Files) {
-        if (F.find("libdevice.") != std::string::npos && F.find(".bc") != std::string::npos) {
-          Out = F;
-          return true;
-        }
+    std::error_code EC;
+    auto Files = common::filesystem::list_regular_files(BitcodeDir, EC);
+    if(EC) return false;
+
+    for(const auto& F : Files) {
+      if (F.find("libdevice.") != std::string::npos && F.find(".bc") != std::string::npos) {
+        Out = F;
+        return true;
       }
-    }catch(...) { /* false will be returned anyway at this point */ }
+    }
```

Comment:
Why do we no longer need the try/catch?

Reply:
Using error code instead of exception handling, as this is compiler world and not runtime world..
See: https://github.com/AdaptiveCpp/AdaptiveCpp/commit/dc965823782910e45837d3cba52764d238a1e839